src/test/java/com/puppycrawl/tools/checkstyle/checks/design/OneTopLevelClassCheckTest.java

```diff
@@ -43,6 +49,27 @@ public void testGetRequiredTokens() {
     }
 
     @Test
+    @SuppressWarnings("unchecked")
+    public void testBeginTreeClearsState() throws Exception {
+        final DetailAST ast = new DetailAST();
+        ast.setType(TokenTypes.CLASS_DEF);
+
+        final DetailAST modifier = new DetailAST();
+        modifier.setType(TokenTypes.MODIFIERS);
+        ast.addChild(modifier);
+
+        final DetailAST child = new DetailAST();
+        child.setType(TokenTypes.IDENT);
+        child.setText("int");
+        ast.addChild(child);
+
+        final OneTopLevelClassCheck check = new OneTopLevelClassCheck();
+        assertTrue("Stateful field is not cleared after beginTree",
+                TestUtils.isStatefulFieldClearedDuringBeginTree(check, ast, "lineNumberTypeMap",
```

Comment:
it is not ok.
It is possible to launch OneTopLevelClassCheck on few files that both have several problems and withtout cleanup of collection second will give unexpected violations.